src/llscan.cc

```
@@ -310,57 +311,76 @@ bool FindReferencesCmd::DoExecute(SBDebugger d, char** cmd,
+
+      SBExpressionOptions options;
+      SBValue value = target.EvaluateExpression(full_cmd.c_str(), options);
+      if (value.GetError().Fail()) {
+        SBStream desc;
+        if (value.GetError().GetDescription(desc)) {
+          result.SetError(desc.GetData());
+        }
+        result.SetStatus(eReturnStatusFailed);
+        return false;
+      }
+      // Check the address we've been given at least looks like a valid object.
+      v8::Value search_value(&llv8, value.GetValueAsSigned());
+      v8::Smi smi(search_value);
+      if (smi.Check()) {
+        result.SetError("Search value is an SMI.");
+        result.SetStatus(eReturnStatusFailed);
+        return false;
+      }
+      scanner = new ReferenceScanner(search_value);
+      break;
+    }
+    case ScanType::kPropertyName: {
+      // TODO - Should I need to check that start++ is null?
+      std::string property_name = *start;
+      scanner = new PropertyScanner(property_name);
+      break;
+    }
+    /* We can add options to the command and further sub-classes of
+     * object scanner to do other searches, e.g.:
+     * - Objects that refer to a particular string literal.
+     *   (lldb) findreferences -s "Hello World!"
+     */
+    case ScanType::kBadOption: {
+      result.SetError("Invalid search type");
+      result.SetStatus(eReturnStatusFailed);
+      return false;
     }
-    result.SetStatus(eReturnStatusFailed);
-    return false;
   }
 
-  // Check the address we've been given at least looks like a valid object.
-  v8::Value search_value(&llv8, value.GetValueAsSigned());
-  v8::Smi smi(search_value);
-  if (smi.Check()) {
-    result.SetError("Search value is an SMI.");
+  /* Ensure we have a map of objects.
+   * (Do this after we've checked the options to avoid
+   * a long pause before reporting an error.)
+   */
+  if (!llscan.ScanHeapForObjects(target, result)) {
+    delete scanner;
```

The main reason for initialising it as part of the options handling was so that we could report any errors in the options before doing `ScanHeapForObjects`which may take quite a while. Otherwise the user sees a huge pause before being told they've entered a bad option.
It only really applies to the first run of the command (if it's the first one that needs to do a scan) as the results of the scan are cached.
I'll merge the change but I'm happy to go back and change it if you think it should. (Just raise an issue.)